src/block_cache/block_chunkcache.c

```diff
@@ -249,6 +249,87 @@ __chunkcache_eviction_thread(void *arg)
     return (WT_THREAD_RET_VALUE);
 }
 
+/*
+ * __chunkcache_str_cmp --
+ *     Qsort function: sort string array.
+ */
+static int WT_CDECL
+__chunkcache_str_cmp(const void *a, const void *b)
+{
+    return (strcmp(*(const char **)a, *(const char **)b));
+}
+
+/*
+ * __chunkcache_arr_free --
+ *     Free the array of strings.
+ */
+static void
+__chunkcache_arr_free(WT_SESSION_IMPL *session, char ***arr)
+{
+    char **p;
+
+    if ((p = (*arr)) != NULL) {
+        for (; *p != NULL; ++p)
+            __wt_free(session, *p);
+        __wt_free(session, *arr);
+    }
+}
+
+/*
+ * __config_gets_pinned_objects --
+ *     Get sorted array of pinned objects from the config.
+ */
+static int
+__config_gets_pinned_objects(WT_SESSION_IMPL *session, const char *cfg[],
+  char ***pinned_objects_list, unsigned int *pinned_entries)
+{
+    WT_CONFIG targetconf;
+    WT_CONFIG_ITEM cval, k, v;
+    WT_DECL_ITEM(tmp);
+    WT_DECL_RET;
+    char **pinned_objects;
+    unsigned int cnt;
+
+    pinned_objects = NULL;
+
+    WT_RET(__wt_config_gets(session, cfg, "chunk_cache.pinned", &cval));
+    __wt_config_subinit(session, &targetconf, &cval);
+    for (cnt = 0; (ret = __wt_config_next(&targetconf, &k, &v)) == 0; ++cnt)
+        ;
+    *pinned_entries = cnt;
+    WT_RET_NOTFOUND_OK(ret);
+
+    if (cnt != 0) {
+        WT_ERR(__wt_scr_alloc(session, 0, &tmp));
```

Comment:
I thought I left a review comment here but I guess not - I think this is fine, the size seems to be more of a hint to avoid reallocations. Loads of other places use 0.